base/strings/util.jl

```diff
@@ -358,7 +358,10 @@ _replace(io, repl, str, r, pattern) = print(io, repl)
 _replace(io, repl::Function, str, r, pattern) =
     print(io, repl(SubString(str, first(r), last(r))))
 
-function replace(str::String, pattern, repl, limit::Integer)
+function replace_new(str::String, pattern, repl, count::Int)
```

Comment:
Ah I see. But this `replace_new` also only accepts a `String`, not an `AbstractString`, so this is the `replace(s::AbstractString, pat, f) = replace_new(String(s), pat, f, typemax(Int))` (still in strings/util.jl, below `replace_new`) which will have to be modified to 
`replace(s::AbstractString, pat, f, count::Integer=typemax(Int)) = replace(String(s), pat, f, clamp(count, typemin(Int), typemax(Int)) % Int)`.